### PR TITLE
DSDEGP-1501 Rip out internal models in prep for adding persistence DAO

### PR DESCRIPTION
**Pull request by danxmoran**
### Description

Resurrection of #94. Description from there was:

Follow-up to #91, not for merging until after it goes through. This deletes Clio's inner models and shifts around the API so that the key+model -> document conversion happens in the search service instead of the search DAO. It should be a no-op chunk of changes that sets up my next PR, which will include the logic for actually writing metadata to GCS.

*Why now?*
To implement the GCS-writing I added a new persistence service / DAO. The "update" function of the search service will move into the persistence service, and will be augmented to first write to storage, then upsert to search. Rather than have both DAOs perform an identical key+value -> document conversion, the service will perform the conversion and pass the resulting document into both DAOs. In prep for making this same flow work across all document types, I pushed the responsibility for defining the converters into the document-specific services, where the transfer -> internal model mappers currently live. With all the mappers in one place, it became pretty clear that the internal models weren't buying us anything, so I scrapped them.

This should also cut down method growth in the search DAO. Instead of defining separate updateWgsUbam, updateCram, etc. methods it just has one update entrypoint (and same for queries).

*How will versioning work now?*
The previous plan for versioning transfer models was to have every transfer model map into the One True internal model. With those internal models gone, we can still implement our versioned transfer models by:

1. Defining a document and query mapper from the latest-versioned transfer models to their corresponding elasticsearch document model.
2. Defining case-class converters from older transfer models to the newest transfer model.

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: Ensure that you have added or updated tests
- [ ] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=114531509).
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Move the JIRA issue to QA once this checklist is completed

### CODE REVIEW

File: clio-server/src/test/scala/org/broadinstitute/clio/server/webservice/WgsUbamWebServiceSpec.scala [hunk 1]
```diff
@@ -133,9 +158,11 @@ class WgsUbamWebServiceSpec
     // We have to test the MemorySearchDAO because we're not going to implement
     // Elasticsearch logic in our test specs. Here, we're just verifying that
     // the web service passes the appropriate queries onto the search DAO.
-    Post("/query", Map("flowcell_barcode" -> "FC123")) ~> webService.queryall ~> check {
```
Comment (danxmoran) [inline clio-server/src/test/scala/org/broadinstitute/clio/server/webservice/WgsUbamWebServiceSpec.scala:136]: While fixing up the `.head` scolding from Codacy I noticed this test was hitting the wrong route.

Comment (tbl3rd) [reply clio-server/src/test/scala/org/broadinstitute/clio/server/webservice/WgsUbamWebServiceSpec.scala:136]: Weird. Why didn't the test realize the URI was wrong?

File: clio-server/src/test/scala/org/broadinstitute/clio/server/webservice/WgsUbamWebServiceSpec.scala [hunk 2]
```diff
@@ -133,10 +158,18 @@ class WgsUbamWebServiceSpec
...
+    Post("/queryall", Map("flowcell_barcode" -> "FC123")) ~> webService.queryall ~> check {
+      memorySearchDAO.queryCalls should be(
+        Seq(
+          // From /query call earlier in the test.
+          TransferWgsUbamV1QueryInput(
+            flowcellBarcode = Some("FC123"),
+            documentStatus = Some(DocumentStatus.Normal)
+          ),
+          // No documentStatus restriction from /queryall
+          TransferWgsUbamV1QueryInput(flowcellBarcode = Some("FC123"))
```
Comment (tbl3rd) [inline clio-server/src/test/scala/org/broadinstitute/clio/server/webservice/WgsUbamWebServiceSpec.scala:170]: When I added a status check to the prior version of the the test, the test failed with `Request was rejected`.

**Review by tbl3rd — APPROVED** [review]
Like the +351 -584, or at least the - part.
Thanks for breaking this review out.

File: clio-server/src/test/scala/org/broadinstitute/clio/server/webservice/WgsUbamWebServiceSpec.scala (hunk 1 above)
Comment (danxmoran) [reply clio-server/src/test/scala/org/broadinstitute/clio/server/webservice/WgsUbamWebServiceSpec.scala:136]: The `check` body was copy-pasted from above, so it was just re-asserting things on the input model from the first `/query` run in the test.

File: clio-server/src/test/scala/org/broadinstitute/clio/server/webservice/WgsUbamWebServiceSpec.scala (hunk 2 above)
Comment (danxmoran) [reply clio-server/src/test/scala/org/broadinstitute/clio/server/webservice/WgsUbamWebServiceSpec.scala:170]: That'd be from the route mismatch in `Post("/query") ~> webService.queryall`.

**Review by pshapiro4broad — APPROVED** [review]

File: clio-server/src/test/scala/org/broadinstitute/clio/server/webservice/WgsUbamWebServiceSpec.scala (hunk 2 above)
Comment (tbl3rd) [reply clio-server/src/test/scala/org/broadinstitute/clio/server/webservice/WgsUbamWebServiceSpec.scala:170]: Yeah, but for the record, even just reporting the status provoked a rejection.
(Maybe check doesn't call awaitResult unless something in the body calls for it?)

**Pull request merged**